README.md

```diff
@@ -1,6 +1,343 @@
+  * `NSLog(@"y: %li", y);`
+  * This should print: `y: -25`.
+
+4. Now override a lower-precedence operation by using parentheses `(` `)` and `NSLog()` the value of `y` again to see how the override affects the result:
+  * `y = 5 - 8 * (4 + 2);`
+  * `NSLog(@"y: %li", y);`
+  * This should print: `y: -43`.
+
+At the end of the code-along, your console should print out something like:
+
+```
+x: 17
+x: 25
+y: -25
+y: -43
+```
+
+### F. Using Floats
+
+1. Declare a new `CGFloat` variable called `f` and set it equal to `0.0`:
+  * `CGFloat f = 0.0;`
+
+2. Write an `NSLog()` that uses the `%f` float format specifier to print `f` to the console along with its variable name:
+  * `NSLog(@"f: %f", f);`
+  * Run the program with `⌘` `R` to see that `f: 0.000000` prints to the console.
+
+3. Redefine `f` to equal the the quotient of dividing `17` by `29`. Copy and paste your `NSLog()` after the redefinition to print the new value to the console:
+  * `f = 17 / 29;`
+  * Run the program with `⌘` `R` to see that `f: 0.000000` still prints to the console. The division operator is returning an integer value as the result of the operation.
+
+4. Redefine `f` with same equation but write at least one of the values as float by using a decimal point and zero:
+  * `f = 17 / 29.0;`
+  * Paste another `NSLog()` and run the program again. You should see `f: 0.586207` printed to the console.
+
+5. Redefine `f` to equal a few more simple calculations. Any time you use the division operator (`/`), define at least one operand as a float value.
+
+```objc
+1 + 2.5
+10 - 1.11
+0.1667 * 6
+3 / 7
+3 / 7.0
+```
+
+If you followed these operations you should see the following lines:
+
+```
+f: 0.000000
+f: 0.000000
+f: 0.586207
+f: 3.500000
+f: 8.890000
+f: 1.000200   // result of 0.1667 * 6
+f: 0.000000   // result of 3 / 7
+f: 0.428571   // result of 3 / 7.0
+```
+
+##### Advanced
+
+If you're feeling adventurous, take a quick dip into the `math.h` library defined in the C-language.
+
+1. Redefine `f` to equal the definition of *pi* `M_PI`, then print `f` again:
+  * `f = M_PI;`
+  * You should see `f: 3.141593` print to the console.
+
+2. But how precise is that definition of *pi*? The default format specifier rounds it to the sixth decimal place. Try using the long-form float format specifier (`%.nf`) to see how many decimals of *pi* that `math.h` knows:
+  * `NSLog(@"f: %.12f", f);`, thought it's more than twelve.
+
+3. Use the `math.h` C-function `sqrt(x)` to calculate the square root of `2`.
+  * `f = sqrt(2);`
+  * How precise is this calculation?
+
+4. Now compare the result of calculating the square root of `2` with `math.h`'s definition of it `M_SQRT2`:
+  * `f = M_SQRT2;`
+  * How does the definition compare to the calculation?
+
+5. Use the `sqrt(x)` function to find the square root of 81.
+
+6. Now use the `pow(x, y)` function to find the cube of 3.
+
+7. Finally, use the `exp2(x)` function to find the highest value of a 64-bit integer.
```

Comment:
yeah i got really tired of this code-along by the end of it, i wasn't thinking things through.